### PR TITLE
More StringMap/StringSet work

### PR DESCRIPTION
**Pull request by arv**
I left these out of the previous commit since these were non trivial.

### CODE REVIEW

**Comment by arv** [comment]
@johnjbarton PTAL

File: src/semantics/VariableBinder.js [hunk 1]
```diff
@@ -39,21 +39,21 @@ import {ScopeChainBuilder} from './ScopeChainBuilder.js';
...
  * @param {boolean=} includeFunctionScope
- * @return {Object}
+ * @return {StringSet}
  */
 export function variablesInBlock(tree, includeFunctionScope = undefined) {
   var builder = new ScopeChainBuilder(null);
   builder.visitAny(tree);
   var scope = builder.getScopeForTree(tree);
-  var names = scope.getLexicalBindingNames();
+  var names = scope.getLexicalBindingNames()
```
Comment (johnjbarton) [inline src/semantics/VariableBinder.js]: trailing ;

File: src/semantics/VariableBinder.js [hunk 2]
```diff
@@ -39,21 +39,21 @@ import {ScopeChainBuilder} from './ScopeChainBuilder.js';
...
   if (!includeFunctionScope) {
     return names;
   }
 
   var variableBindingNames = scope.getVariableBindingNames();
-  for (var name in variableBindingNames) {
-    names[name] = true;
-  }
+  variableBindingNames.forEach((name) => {
```
Comment (johnjbarton) [inline src/semantics/VariableBinder.js]: probably could be on a single line

**Comment by johnjbarton** [comment]
LGTM

File: src/semantics/VariableBinder.js (hunk 2 above)
Comment (arv) [inline src/semantics/VariableBinder.js]: Done. Changed the other 2 cases in this PR too.

File: src/semantics/VariableBinder.js (hunk 1 above)
Comment (arv) [inline src/semantics/VariableBinder.js]: Done.

**Pull request merged**